### ISSUE TITLE
Add pathways function

### ISSUE THREAD

**Issue by karissawhiting** [MSKCC-Epi-Bio/gnomeR#154]
I removed the pathways argument from the binary matrix function, but we still have the `custom_pathway()` function. I'm thinking we should remove this as well and have one unified function called `add_pathway()` that can annotate pre-set pathways included in the package, or allow you to pass a custom pathway. It could be used as follows:

``` r
binary_matrix(mutation = mut, cna = cna,  fusions = fus) %>%
    add_pathway(pathway = "all")
```
Where the `pathway` argument would be "all" be default, allowing you to annotate all package-included pathways. (see `gnomeR::pathways`). Otherwise you could specify a specific `pathway_id` like "RTK-RAS". 

``` r
binary_matrix(mutation = mut, cna = cna,  fusions = fus) %>%
    add_pathway(pathway = "RTK-RAS")
```

Additionally, you could add a custom pathway with:

``` r
binary_matrix(mutation = mut, cna = cna,  fusions = fus) %>%
    add_pathway(custom_pathway = c("TP53", "APC", "FGFR3"))
```

First step of this will be updating the `gnomeR::pathways` internal data which is currently incorrect. See [pathways.R](https://github.com/MSKCC-Epi-Bio/gnomeR/blob/master/data-raw/pathways.R) which pulls data from [pathways.csv](https://github.com/MSKCC-Epi-Bio/gnomeR/blob/master/data-raw/pathways.csv) to create this internal data frame. You can update pathways.csv directly.

**Comment by karissawhiting** [comment]
@carokos let me know if you are interested in working on this, thanks!

**Comment by karissawhiting** [comment]
Need to test and add custom pathway function.

**Comment by karissawhiting** [comment]
@carokos I expanded this function a bit, allowing for custom pathways and also allowing annotation by gene or specific alteration. Please give it a test run if you can.

Thanks for your work on this! <3

**Comment by carokos** [comment]
I found a typo in the file I was using with the possible pathway alterations.  When I get a minute, I'll update and push the corrected file.

**Comment by karissawhiting** [comment]
closed by #163
